user_guide.adoc

```diff
@@ -1377,30 +1377,82 @@ library" for expression language.
 |Return type
 |Description
 
-|to_s
+|`to_s`
 |String
 |Converts integer into a string using decimal representation
-
-|to_s(radix)
-|String
-|Converts integer into a string using specified `radix` (i.e. use `16` to get hexadecimal representation, use `8` to get octal, etc)
 |===
 
 ==== Floating point numbers
 
-TODO
+No methods are currently supported for floating point numbers.
 
 ==== Byte arrays
 
-TODO
+[cols="3*", options="header"]
+|===
+|Method name
+|Return type
+|Description
+
+|`to_s(encoding)`
+|String
+|Decodes (converts) a byte array encoded using the specified `encoding` scheme into a string
+|===
 
 ==== Strings
 
-TODO
+[cols="3*", options="header"]
+|===
+|Method name
+|Return type
+|Description
+
+|`length`
+|Integer
+|Length of a string in number of characters
+
+|`reverse`
+|String
+|Reversed version of a string
+
+|`substring(from, to)`
+|String
+|Extracts a portion of a string between character at offset `from` and character at offset `to` (inclusive of characters at `from` and `to` offsets)
```

Comment:
`foo.substring(1, 3)` in KSY translates to:

* `foo().substr(1, (3) - (1));` in C++
* `Foo.Substring(1, 3 - 1)` in C#
* `foo().substring(1, 3)` in Java
* `this.foo.substring(1, 3)` in JS
* `$self->foo()[1:3]` in Perl
* `foo[1, (3 - 1)]` in Ruby

I believe it all yields a string starting from character with index=1 (first character has index=0), spanning two characters, i.e. "from" index is included, and "to" is excluded.